### PR TITLE
feat: refactor database cli push to use buildx

### PR DESCRIPTION
**Pull request by xavidop**
**Fixes or implements IN-000**

### Brief description. What is this change?

This PR adds the following changes:
1. Refactor update database track job to use buildx
2. Add the ability to push or not a docker image using build
3. Add semantic override for database cli
4. fixes an issue with provision-env when force creating an environment

### Checklist

- [ ] Breaking changes have been communicated, including:
    - New required environment variables
    - Renaming of interfaces (API routes, request/response interface, etc)
- [ ] New environment variables have [been deployed](https://www.notion.so/voiceflow/Add-Environment-Variables-be1b0136479f45f1adece7995a7adbfb)
- [ ] Appropriate tests have been written
    - Bug fixes are accompanied by an updated or new test
    - New features are accompanied by a new test

### CODE REVIEW

**Comment by xavidop** [comment]
bors try

File: src/commands/vfcli/vfcli-delete-or-release-env.yml [hunk 1]
```diff
@@ -36,6 +40,24 @@ steps:
...
+                --data "{\"parameters\":{\"e2e_env_name\":\"$env_name\",\"trigger_pipeline\":true}}")
+              if [[ $response == "201" ]]; then
+                echo "Env Release Pipeline triggered successfully. Exiting with success."
+                exit 0
+              else
+                echo "Failed to trigger pipeline. HTTP status code: $response"
+                exit 1
+              fi
+            fi
             vfcli env resume "$env_name" --interactive false
```
Comment (edison-vflow) [inline src/commands/vfcli/vfcli-delete-or-release-env.yml:61]: We resume env_name after receiving a 201.Does 201 mean he pipeline finished or it was successfully triggered.
If its a successful trigger only, will resuming env while database is resetting cause some weird state ?

Comment (xavidop) [reply src/commands/vfcli/vfcli-delete-or-release-env.yml:61]: oof, that is part of another PR, let me remove it

File: src/commands/vfcli/vfcli-create-or-get-free-env-from-pool.yml [hunk 2]
```diff
@@ -29,12 +29,15 @@ steps:
...
       name: Check for free environment or Create environment
       command: |
+        force="<< parameters.force >>"
         # fetches a free environment from the pool if pool-type is provided
         # otherwise creates a new environment with the provided name
         # and stores the env name in the cache
         if [[ -n "<< parameters.pool-type >>" ]]; then
           if [[ $force ]]; then
             echo "Force option is enabled... Proceeding with environment creation."
+            echo "null" > env_name.txt  # Write 'null' to indicate no environment was found
```
Comment (edison-vflow) [inline src/commands/vfcli/vfcli-create-or-get-free-env-from-pool.yml:39]: I think we always expect an environment to be available, either in the parameter or in the file.
Can all logic in the ORB handle having a null as env name in the env_name.txt ?
This is extensively used across our ORBs to determine the environment to use 
cc @junydania

**Review by edison-vflow — COMMENTED** [review]
Happy to approve after comments are answered

File: src/commands/vfcli/vfcli-create-or-get-free-env-from-pool.yml (hunk 2 above)
Comment (xavidop) [reply src/commands/vfcli/vfcli-create-or-get-free-env-from-pool.yml:39]: yeah, it is built to handle that

**Pull request closed without merging**